force-app/main/default/lwc/serviceDeliveryRow/serviceDeliveryRow.js

```diff
@@ -264,9 +265,13 @@ export default class ServiceDeliveryRow extends LightningElement {
         this.handleSaveEnd();
         this.lockContactField();
         fireEvent(this.pageRef, "serviceDeliveryUpsert", event.detail);
-        if (event.detail.fields[this.fields.unitOfMeasurement.fieldApiName]) {
+        if (
+            event.detail.fields[this.fields.unitOfMeasurement.fieldApiName].value !== null
+        ) {
             this.unitOfMeasureValue =
                 event.detail.fields[this.fields.unitOfMeasurement.fieldApiName].value;
+        } else if (this.unitOfMeasureValue !== this.labels.quantity) {
```

Comment:
Could this not just be an else case?  Guessing we'd want the value to be quantity anytime the actual UoM isn't present.

Reply:
@AndersonTarren , The else if is to set the value back to the default if they change from a service with a UoM to one without a UoM